src/number_theory/primorial.lean

```diff
@@ -5,24 +5,29 @@ open_locale big_operators
 open finset
 open nat
 
+def primorial (n : ℕ) : ℕ := ∏ p in (filter prime (range (n + 1))), p
+notation x`#` := primorial x
+
 -- This is going into stdlib
 lemma sum_range_choose_halfway (m : ℕ) : ∑ i in range (m + 1), choose (2 * m + 1) i = 4 ^ m :=
    by sorry
 
 private lemma choose_symm_half (m : ℕ) : choose (2 * m + 1) (m + 1) = choose (2 * m + 1) m :=
 by apply choose_symm_of_eq_add; ring
 
-lemma extend_prime_product_odd (n : ℕ) (n_big : 2 < n) (r : n % 2 = 0) :
-  ∏ p in (filter prime (range n)), p = ∏ p in (filter prime (range (n + 1))), p
-  :=
+lemma primorial_succ (n : ℕ) (n_big : 2 < n) (r : n % 2 = 1) : n# = (n + 1)# :=
 begin
-  have not_prime : ¬prime n,
+  have not_prime : ¬prime (n + 1),
     { intros is_prime,
       cases (prime.eq_two_or_odd is_prime),
-      { subst h, linarith, },
-      { linarith }, },
+      { linarith, },
+      { rw ←not_even_iff at h r,
```

Comment:
Why did this proof become so long now? Is this because of the definition `primorial`? In that case, use `unfold primorial` to get rid of it inside this proof.

Reply:
No, it's because I flipped the parity of the inputs, with the result that the parity condition no longer gives me something I can `subst`.

Reply:
So maybe the proof that you had before was "better"? What was the reason for switching the inputs? Some painpoint somewhere else?

Reply:
The theorem I had before couldn't be stated in terms of `primorial`. It was essentially `(n - 1)# = n#`, which is very sad. The theorem really "wants" to be `n# = (n + 1)#`, which is what it now is.

Reply:
Ok, I see. Maybe a linarith expert can take a look at this in the near future.